ui/pages/keychains/reveal-seed.js

```diff
@@ -1,255 +1,304 @@
       passwordBox.focus();
     }
-  }
+  }, []);
 
-  handleSubmit(event) {
+  const renderQR = () => {
+    const qrImage = qrCode(0, 'L');
+    qrImage.addData(seedWords);
+    qrImage.make();
+    return qrImage;
+  };
+
+  const handleSubmit = (event) => {
     event.preventDefault();
-    this.setState({ seedWords: null, error: null });
-    this.props
-      .requestRevealSeedWords(this.state.password)
-      .then((seedWords) => {
-        this.context.trackEvent({
+    setSeedWords(null);
+    setCompletedLongPress(false);
+    setError(null);
+    try {
+      dispatch(requestRevealSeedWords(password)).then((revealedSeedWords) => {
+        trackEvent({
           category: EVENT.CATEGORIES.KEYS,
           event: EVENT_NAMES.KEY_EXPORT_REVEALED,
           properties: {
             key_type: EVENT.KEY_TYPES.SRP,
           },
         });
-        this.setState({ seedWords, screen: REVEAL_SEED_SCREEN });
-      })
-      .catch((error) => {
-        this.context.trackEvent({
-          category: EVENT.CATEGORIES.KEYS,
-          event: EVENT_NAMES.KEY_EXPORT_FAILED,
-          properties: {
-            key_type: EVENT.KEY_TYPES.SRP,
-            reason: error.message, // 'incorrect_password',
+        setSeedWords(revealedSeedWords);
+      });
+      dispatch(
+        showModal({
+          name: 'HOLD_TO_REVEAL_SRP',
+          onLongPressed: () => {
+            setCompletedLongPress(true);
+            setScreen(REVEAL_SEED_SCREEN);
           },
-        });
-        this.setState({ error: error.message });
+        }),
+      );
+    } catch (e) {
+      trackEvent({
+        category: EVENT.CATEGORIES.KEYS,
+        event: EVENT_NAMES.KEY_EXPORT_FAILED,
+        properties: {
+          key_type: EVENT.KEY_TYPES.SRP,
+          reason: e.message, // 'incorrect_password',
+        },
       });
-  }
+      setError(e.message);
+    }
+  };
 
-  renderWarning() {
+  const renderWarning = () => {
     return (
-      <div className="page-container__warning-container">
-        <i className="fa fa-exclamation-triangle fa-2x page-container__warning-icon" />
-        <div className="page-container__warning-message">
-          <div className="page-container__warning-title">
-            {this.context.t('revealSeedWordsWarningTitle')}
-          </div>
-          <div>{this.context.t('revealSeedWordsWarning')}</div>
+      <div className="srp__warning-container">
+        <div className="srp__warning-icon">
+          <Icon name="warning-filled" color={COLORS.ERROR_DEFAULT} />
+        </div>
+        <div className="srp__warning-message">
+          <Text variant={TEXT.BODY_MD}>
+            {t('revealSeedWordsWarning', [
+              <Text
+                key="reveal-seed-words-warning-2"
+                variant={TEXT.BODY_MD_BOLD}
+                as="span"
+              >
+                {t('revealSeedWordsWarning2')}
+              </Text>,
+            ])}
+          </Text>
         </div>
       </div>
```

Comment:
Should be able to use the recently merged `Banner` component here. Docs 👇   https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-banner--default-story#severity

Reply:
thanks :+1: